manager/network/model.go

```diff
@@ -0,0 +1,15 @@
+package network
+
+import (
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/allocator/networkallocator"
+)
+
+// Model is an abstraction over the Network Model to be used.
+type Model interface {
+	Allocator() (networkallocator.NetworkAllocator, error)
+	SetDefaults(spec *api.NetworkSpec) error
+	ValidateNetworkSpec(spec *api.NetworkSpec) error
+	PredefinedNetworks() []networkallocator.PredefinedNetworkData
```

Comment:
@aaronlehmann done. I also refactored the rest of `manager/allocator/networkallocator` and a few bits of `manager/allocator/network.go` out into various more suitable places, so the packages which depend on the allocator are now (IMHO at least) much cleaner.